apps/explorer/app/contract/[id]/page.tsx

```diff
@@ -42,44 +38,58 @@ export default async function Page({ params }) {
     ),
   ])
 
-  if (error) {
-    throw error
-  }
+  if (rateError) throw rateError
+  if (!rate) return notFound()
 
-  const contract = c?.contract
+  // Grab the contract and previous revisions data.
+  const [
+    [contract, contractError],
+    [previousRevisions, previousRevisionsError],
+  ] = await Promise.all([
+    to(explored.contractByID({ params: { id } })),
+    to<
+      FileContractElement[] // Not sure why this is necessary here.
+    >(explored.contractRevisions({ params: { id } })),
+  ])
 
-  if (!contract) {
-    return notFound()
-  }
+  if (contractError) throw contractError
+  if (previousRevisionsError) throw previousRevisionsError
+  if (!contract || !previousRevisions) return notFound()
 
-  const formationTxnId = getFormationTxnId(contract)
-  const finalRevisionTxnId = contract?.transaction_id || ''
+  // What should we do here? What's going on now is wrong, I'm sure, and
+  // lots of things below here in the file and children hinge on this.
+  const formationTxnId = getFormationTxnId(contract, previousRevisions)
+  const finalRevisionTxnId = contract.confirmationTransactionID || ''
 
-  const [[ft], [rt]] = await Promise.all([
+  // Fetch our formation and finalRevision transactions
+  const [
+    [formationTransaction, formationTransactionError],
+    [renewalTransaction, renewalTransactionError],
+  ] = await Promise.all([
     to(
-      siaCentral.transaction({
+      explored.transactionByID({
         params: {
           id: formationTxnId,
         },
       })
     ),
     to(
-      siaCentral.transaction({
+      explored.transactionByID({
         params: {
           id: finalRevisionTxnId,
         },
       })
     ),
   ])
 
-  const formationTransaction = ft?.transaction
-  const renewedFrom = formationTransaction?.contract_revisions?.[0]
-  const renewalTransaction = rt?.transaction
-  const renewedTo = renewalTransaction?.storage_contracts?.[0]
+  if (formationTransactionError) throw formationTransactionError
+  if (renewalTransactionError) throw renewalTransactionError
+  if (!formationTransaction || !renewalTransaction) return notFound()
+
+  const renewedFrom = formationTransaction.fileContractRevisions[0]
+  const renewedTo = renewalTransaction.fileContracts[0]
 
-  // The following is a temporary addition to satisfy new Transaction
-  // component requirements for the Sia Central phase out on the
-  // transaction route.
+  // Do our transaction fetching, chain indices, and tip fetching.
   const [
     [transaction, transactionError],
     [transactionChainIndices, transactionChainIndicesError],
```

Comment:
<!--__GRAPHITE_HTML_TAG_START__--><i class='graphite__hidden'>[Re: line +99]<br/><br/></i>

<!--__GRAPHITE_HTML_TAG_END__-->I believe this is refetching a txn we already have.<!--__GRAPHITE_HTML_TAG_START__--><span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/SiaFoundation/web/761?utm_source=unchanged-line-comment">Graphite</a>.</span>

<!--__GRAPHITE_HTML_TAG_END__-->

Reply:
Leaving this open before I submit this PR because the issue is that we're not guaranteed confirmationIndex. It can be null.